### PR TITLE
DeveloperGuide: update implementation for 'PlannerList'

### PR DESCRIPTION
**Pull request by Letm3through**
Let's update DeveloperGuide to include 'PlannerList' which is to
be updated in the future.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+=== Planner List Feature
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc:387]: Could you add a tag in front of this section eg: `// tag::planner-list[]`

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
+returns degree planner(s) having its year matches the year given
+
+* list degree planner(s) by semester
+i.e. `planner_list s/SEMESTER`
+returns degree planner(s) having its semester matches the semester given
+
+* include `year` and `semester` attributes in one `planner_list` command and list degree planner(s)
+i.e. `planner_list y/YEAR s/SEMESTER`
+returns module having its year or semester matches the given year and semester
+
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc:420]: Could you add a tag at the end of this section as well?
eg: // end::planner-list[]

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
+are taken by the program.
+
+1. Extract out the text related to `planner_list` command
+2. Parse the text related to each `PREFIX` individually.
+3. Return a composite predicate for all attributes.
+
+.PlannerList component interactions
+image::PlannerListComponentSequenceDiagram.png[width="650"]
+
+==== Current Implementation
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc:405]: You might want to consider adding in some design considerations as well

**Review by chyeo — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (Letm3through) [reply docs/DeveloperGuide.adoc:387]: Working on the part. Thanks for mentioning.

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (truegitnovice) [reply docs/DeveloperGuide.adoc]: Added one for now. Thank you

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -383,6 +383,49 @@ In addition to adding module code(s) to a requirement category, the `Requirement
...
+returns degree planner(s) having its semester matches the semester given
+
+* include `year` and `semester` attributes in one `planner_list` command and list degree planner(s)
+i.e. `planner_list y/YEAR s/SEMESTER`
+returns module having its year or semester matches the given year and semester
+
+==== Design Considerations
+===== Aspect: How to parse multiple attributes
+
+- Alternative 1 (current choice): Parse the text related to each `PREFIX` individually
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc:424]: This `Alternative 1` is it compared to anything that lead you to chose this over the other?
I think that including the other alternatives would give the reader a better idea of why `Alternative 1` was chosen.

**Review by chyeo — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc (hunk 4 above)
Comment (Letm3through) [reply docs/DeveloperGuide.adoc:424]: Thanks for the pinpoint. Added additional alternative.

**Review by chyeo — APPROVED** [review]
LGTM. Thanks for updating the developer guide!

**Pull request merged**